### ISSUE TITLE
Metric Evaluation

### ISSUE THREAD

**Issue by Hao-FANG-92** [woodfrog/floor-sp#11]
Hi,

Thanks for sharing this outstanding work with the public.

Actually, I wonder how to evaluate the four metrics mentioned in the paper if we got the floorplan ?

**Comment by woodfrog** [comment]
Hi,

Thank you for your interest.

We used the [evaluation code of FloorNet](https://github.com/art-programmer/FloorNet/blob/master/evaluate.py) to get the numbers in the paper (with some simple modification for our output format).

**Comment by Hao-FANG-92** [comment]
> Hi,
> 
> Thank you for your interest.
> 
> We used the [evaluation code of FloorNet](https://github.com/art-programmer/FloorNet/blob/master/evaluate.py) to get the numbers in the paper (with some simple modification for our output format).

Thanks for the reply!

I wonder if there is any relationship between the four metrics and the functions in utils.py, i.e. evaluateSegmentation, evaluateCorners, etc ? If not, it means that we can get the 4 metrics using the evaluation code of FloorNet ?

Best!

**Comment by woodfrog** [comment]
Uh, sorry for the confusion. The file utils.py was directly copied from FloorNet's codebase... 

FloorNet provides the functions for computing recall/precision for walls(edges) and rooms, these functions can be adapted to compute the other two metrics (the basic logic for recall/precision stays the same).

**Comment by Hao-FANG-92** [comment]
> Uh, sorry for the confusion. The file utils.py was directly copied from FloorNet's codebase...
> 
> FloorNet provides the functions for computing recall/precision for walls(edges) and rooms, these functions can be adapted to compute the other two metrics (the basic logic for recall/precision stays the same).

Thansk for the response.

I am trying to figure out how to evaluate the trained model on the provided dataset. Could you please share the pretrained model of FloorNet on the dataset of FloorSP ?

**Comment by woodfrog** [comment]
> > Uh, sorry for the confusion. The file utils.py was directly copied from FloorNet's codebase...
> > FloorNet provides the functions for computing recall/precision for walls(edges) and rooms, these functions can be adapted to compute the other two metrics (the basic logic for recall/precision stays the same).
> 
> Thansk for the response.
> 
> I am trying to figure out how to evaluate the trained model on the provided dataset. Could you please share the pretrained model of FloorNet on the dataset of FloorSP ?

Sorry, I didn't back up the pretrained FloorNet (and the modified evaluation code) of Beike data myself. They should still be on my previous school server but I don't have access to them now.  For evaluation, you might need to read the evaluation logics in FloorNet's repo and try to adapt it yourself (some other people did similar things, the numbers were not exactly the same but almost match the ones in the paper).  Sorry for the inconvenience.

**Comment by Hao-FANG-92** [comment]
> > > Uh, sorry for the confusion. The file utils.py was directly copied from FloorNet's codebase...
> > > FloorNet provides the functions for computing recall/precision for walls(edges) and rooms, these functions can be adapted to compute the other two metrics (the basic logic for recall/precision stays the same).
> > 
> > 
> > Thansk for the response.
> > I am trying to figure out how to evaluate the trained model on the provided dataset. Could you please share the pretrained model of FloorNet on the dataset of FloorSP ?
> 
> Sorry, I didn't back up the pretrained FloorNet (and the modified evaluation code) of Beike data myself. They should still be on my previous school server but I don't have access to them now. For evaluation, you might need to read the evaluation logics in FloorNet's repo and try to adapt it yourself (some other people did similar things, the numbers were not exactly the same but almost match the ones in the paper). Sorry for the inconvenience.

Thanks a lot anyway !

I am a bit confused about the evaluation code of FloorNet. Maybe I'd better rewrite the evaluation code by myself which could be save a bit of time.

Best.

**Comment by chinhsuanwu** [comment]
Hi @Hao-FANG-92 

Have you figured out the evaluation metrics already?
Can you please share it with me if possible? I will appreciate it.
